crates/bevy_ecs/src/change_detection.rs

```diff
@@ -284,6 +337,56 @@ impl<'a> Ticks<'a> {
     }
 }
 
+/// Shared borrow of an entity's component with change tracking metadata
+pub struct Ref<'a, T: ?Sized> {
+    pub(crate) value: &'a T,
+    pub(crate) ticks: Ticks<'a>,
+}
+
+impl<'w, 'a, T> IntoIterator for &'a Ref<'w, T>
+where
+    &'a T: IntoIterator,
+{
+    type Item = <&'a T as IntoIterator>::Item;
+    type IntoIter = <&'a T as IntoIterator>::IntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.value.into_iter()
+    }
+}
+change_detection_impl!(Ref<'a, T>, T,);
+impl_debug!(Ref<'a, T>,);
+
+
+/// Shared borrow of a [`Resource`] with change tracking metadata.
+///
+/// See the [`Resource`] documentation for usage.
+///
+/// # Panics
+///
+/// Panics when used as a [`SystemParam`](crate::system::SystemParam) if the resource does not exist.
+///
+/// Use `Option<ResRef<T>>` instead if the resource might not always exist.
+pub struct ResRef<'a, T: ?Sized + Resource> {
```

Comment:
`Res<T>` already has change tracking metadata, is this type necessary?

Reply:
Agreed.